### PR TITLE
Revoke invitation #160300315

### PR DESCRIPTION
**Pull request by dandds**
https://www.pivotaltracker.com/story/show/160300315

This gives workspace owners and admins the ability to revoke a pending invitation.

I also added a new macro for a confirmation button component. The button presupposes that it's being used as the D in CRUD, so it's wrapped in its own small `form` element and needs a form action specified. This should prevent any form inheritance and submission confusion in the DOM.

The confirmation popup and copy could use a UI/UX pass if you have time, @luiscielak or @andrewcroce .

screen shot 2018-11-07 at 9 46 49 am

### CODE REVIEW

File: nohup.out [hunk 1]
```diff
@@ -0,0 +1,7 @@
+3296:C 07 Nov 08:57:25.261 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
```
Comment (richard-dds) [inline nohup.out]: You probably didn't mean to commit this

File: atst/routes/workspaces.py [hunk 2]
```diff
@@ -357,10 +357,24 @@ def update_member(workspace_id, member_id):
...
 
     return redirect(
         url_for("workspaces.show_workspace", workspace_id=invite.workspace.id)
     )
+
+
+@bp.route("/workspaces/<workspace_id>/invitations/<token>/revoke", methods=["POST"])
+def revoke_invitation(workspace_id, token):
+    workspace = Workspaces.get(g.current_user, workspace_id)
+    Authorization.check_workspace_permission(
```
Comment (richard-dds) [inline atst/routes/workspaces.py]: I feel like we've done this in a few places recently, but I thought that we'd decided to do auth in the domain layer?

Comment (dandds) [reply atst/routes/workspaces.py]: You're right. I was following the existing route functions in that module, which all do it directly. I think that whole module needs to be refactored.

File: nohup.out (hunk 1 above)
Comment (dandds) [reply nohup.out]: sure didn't, thanks

**Comment by dandds** [comment]
@richard-dds Added another Workspace domain method for getting a workspace in order to update its members. I definitely prefer having authz logic in the domain. At the same time, it seems like we're adding a lot of "get this resource for purpose X" methods, which are just thin wrappers so that the caller doesn't have to know a specific permission name. Maybe that's fine, but it doesn't sit well with me. I'd like to see if we can hit on a better, maybe more generic, way of handling this.

**Comment by richard-dds** [comment]
@dandds I definitely see what you mean, and I don't really like it either. Let's think / talk about this

**Pull request merged**